File: Java Labs/Patterns On Java/src/com/pattern/mediator/OrderProcessingSystem.java
package com.pattern.mediator;

public class OrderProcessingSystem implements Component {
    private Mediator mediator;

    @Override
    public void setMediator(Mediator mediator) {
        this.mediator = mediator;
    }

    @Override
    public void notify(String event) {
        System.out.println("Order Processing System received notification: " + event);
        // Additional logic related to order processing...
    }

    public void placeOrder(String item, int quantity) {
        // Order processing logic...
        mediator.notify(this, "Order placed for item: " + item + ", Quantity: " + quantity);
    }
}

File: Python Labs/com/pattern/Mediator/OrderProcessingSystem.py
from com.pattern.Mediator.Component import Component


class OrderProcessingSystem(Component):
    def __init__(self):
        self.mediator = None

    def set_mediator(self, mediator):
        self.mediator = mediator

    def notify(self, event):
        print(f"Order Processing System received notification: {event}")
        # Additional logic related to order processing...

    def place_order(self, item, quantity):
        # Order processing logic...
        self.mediator.notify(self, f"Order placed for item: {item}, Quantity: {quantity}")